### PR TITLE
Fix warg update

### PR DESCRIPTION
**Pull request by lann**

### CODE REVIEW

File: crates/client/src/storage/fs.rs [hunk 1]
```diff
@@ -95,11 +96,12 @@ impl RegistryStorage for FileSystemRegistryStorage {
     async fn load_packages(&self) -> Result<Vec<PackageInfo>> {
         let mut packages = Vec::new();
 
-        for entry in WalkDir::new(&self.base_dir) {
+        let packages_dir = self.base_dir.join(PACKAGE_LOGS_DIR);
+        for entry in WalkDir::new(&packages_dir) {
```
Comment (lann) [inline crates/client/src/storage/fs.rs:100]: The minimal fix for `warg update` would have been

```suggestion
        for entry in WalkDir::new(&packages_dir.join("sha256")) {
```

but given how packages are iterated over this seemed more robust.

File: crates/client/src/storage/fs.rs [hunk 2]
```diff
@@ -70,10 +71,10 @@ impl FileSystemRegistryStorage {
     }
 
     fn package_path(&self, id: &PackageId) -> PathBuf {
-        self.base_dir.join(
+        self.base_dir.join(PACKAGE_LOGS_DIR).join(
             LogId::package_log::<Sha256>(id)
                 .to_string()
-                .replace(':', "/"),
+                .replace(':', "-"),
```
Comment (peterhuene) [inline crates/client/src/storage/fs.rs:77]: As `WalkDir` is recursive, can we keep the algo directory?

**Review by peterhuene — APPROVED** [review]

**Comment by peterhuene** [comment]
Also probably need a client test to cover the update command?

File: crates/client/src/storage/fs.rs (hunk 2 above)
Comment (lann) [reply crates/client/src/storage/fs.rs]: Sure.

**Comment by lann** [comment]
Added an assertion to the client megatest & confirmed it would have caught the regression.

**Review by peterhuene — APPROVED** [review]

**Pull request merged**